### PR TITLE
Add Circle CI config

### PR DESCRIPTION
**Pull request by shigmas**
Adding Circle CI to the v3_baseline, since it tests v2 and v3 in different jobs, and master doesn't have v3 yet.

### CODE REVIEW

**Comment by bananaumai** [comment]
Can you enable circleci build? Then you can see if the build succeeds.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+version: 2.0
+jobs:
+  go-vet-and-test:
+    docker:
+      - image: circleci/golang:1.13.3
+    working_directory: /go/src/github.com/moderepo/device-sdk-go
```
Comment (bananaumai) [inline .circleci/config.yml]: I'm not so sure, but we may not have to specify working_directory because we now are using go module.

Comment (shigmas) [reply .circleci/config.yml]: Cool. Removed it and it ran locally fine.

**Review by bananaumai — APPROVED** [review]
👍

**Pull request merged**